# Homeworks

In [2]:
import pandas as pd

In [5]:
df= pd.read_csv("data.csv")

In [8]:
df

พนักงาน  JP   IQ
0  พนักงาน1   7   99
1  พนักงาน2   9  105
2  พนักงาน3  12  113
3  พนักงาน4  14  115
4  พนักงาน5  16  118

Predict JP values 
JP is a independent variable and IQ is a dependent variable 

In [18]:
X = df.IQ.values.reshape((-1,1))
y = df.JP.values

In [19]:
import numpy as np
from sklearn.linear_model import LinearRegression

In [20]:
model = LinearRegression()
model.fit(X,y)

LinearRegression()

In [21]:
a = model.coef_
b = model.intercept_

In [22]:
a

array([0.45901639])

In [23]:
b

-38.89180327868853

###### Equation y = ax+c --> y = -38.89 x + 0.46

###### หากพนักงานคนนึงมี IQ = 110เขาจะมีคะแนนJPเท่าใด

In [28]:
y_pred = model.predict(np.array([[110]]))
# answer
y_pred

array([11.6])

In [29]:
# MSE RMSE
from sklearn import metrics

In [30]:
y_pred = model.predict(X)

In [31]:
MSE = metrics.mean_squared_error(y,y_pred)
RMSE = np.sqrt(MSE)

In [32]:
MSE

0.3580327868852466

In [34]:
RMSE

0.5983584100564198

### Answer
1.IQ <br>
2.JP <br>
3.y = -38.89 x + 0.46 <br>
4.11.6 <br>
5.MSE = 0.3580327868852466 <br>
  RMSE = 0.5983584100564198 <br>